plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -103,8 +107,12 @@ class session_manager {
    void pop_entry(bool call_send = true) {
       send_queue.erase(send_queue.begin());
       sending = false;
-      if (call_send || !send_queue.empty())
-         send();
+      if (call_send || !send_queue.empty()) {
+         // avoid blowing the stack
+         boost::asio::post(ship_io_context, [this]() {
```

Comment:
Is it easy to add a unit test to do a relative large number of recursive sends and it does not blow up?

Reply:
I tried a few diff approaches but was unable to reproduce in a test env.

Reply:
That's OK. Code inspection would be sufficient for this case.